test/compound/protocol.test.ts

```diff
@@ -29,6 +27,13 @@ describe("Compound ", () => {
     expect(() => new Compound({})).toThrowError("invalid web3Provider");
   });
 
+  it("Should throw error because provider was already given", async () => {
+    const protocol = new Compound(ethereumObject);
+    expect(() => protocol.updateProvider(ethereumObject)).toThrowError(
+      "Provider already instanciated"
+    );
+  });
+
   it("New contract instance ", () => {
     const protocol = new Compound(ethereumObject);
     const contractInstance = protocol.getContract(
```

Comment:
This seems weird to me to have to specify the ABI directoy. Why not have the user give a name of a know ABI in the protocol. This doesn't seem very app developer friendly.

I would think that you should be able to do something like this:
```ts
const compound = new Compound(config)
const governor = compound.governorAlpha()
governor.createProposal()
governor.vote()
governor.proposals()
etc...
```

Reply:
Dude thanks for taking the time of checking this out 🙌 

I was thinking the same, the contracts ABI are now hardcoded and if you want to update them then you'd need to update the SDK code. 

Right now the user can create an instance of the SDK by passing the `web3.currentProvider` object or an url (to just make calls), maybe we can add also an optional argument in the Compound class [constructor](https://github.com/dOrgTech/compound-sdk/blob/implementing_contract_methods/src/compound/index.ts#L18) that can be the ABI of the contracts?

Also, just to make things clear, the user will do exactly what you just said - but as I said before, the `config` variable is a `web3provider` or `string`:
>const compound = new Compound(config)
const governor = compound.governorAlpha()
governor.createProposal()
governor.vote()
governor.proposals()
etc...

I was thinking to apply a logic like: If user pass ABIs in constructor arguments, we use those, otherwise the ones that are hardcoded, does this makes sense? @dOrgJelli

Reply:
> maybe we can add also an optional argument in the Compound class constructor that can be the ABI of the contracts?

I don't know if this would be a common use case. I think we should pull in the contract ABIs from a single source of truth (if one exists). I think the github release is a good place.

Reply:
It's done sir! Thank you for the recommendation :-) @dOrgJelli